File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void hash(int hash);
void space(int space);


int main(void)
{

    int height;

    do
    {
        height = get_int("Height: ");//prompt user for valid input
    }
    while (height < 1 || height > 8);

    int no_spaces = height - 1;
    int no_hash = 1;

    for (int i = 0; i < height; i++)//for each row
    {
        space(no_spaces);//print space
        hash(no_hash);//print left aligned hashes

        printf("  ");
        hash(no_hash); //print right aligned hashes
        printf("\n");

        no_spaces--; //decrement number of spaces
        no_hash++; //incrememnt number of hashes
    }
}

void hash(int hash)
{
    int i = 0;
    while (i < hash)
    {
        printf("#");
        i++;
    }
}

void space(int space)
{
    int i = 0;
    while (i < space)
    {
        printf(" ");
        i++;
    }
}

File: sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = 0

    # prompt user until user enters valid input
    while height < 1 or height > 8:
        height = get_int("Height: ")

    # initialize number of hashes and spaces to be printed on row 1
    no_hash = 1
    no_spaces = height - 1

    for i in range(height):
        spaces(no_spaces)  # print spaces
        hash(no_hash)  # print left hashes

        print("  ", end="")  # print space between left and right hashes
        hash(no_hash)  # print right hashes
        print()  # print new line

        no_spaces -= 1
        no_hash += 1


def spaces(space):
    for i in range(space):
        print(" ", end="")


def hash(hash):
    for i in range(hash):
        print("#", end="")


main()